metals/src/main/scala/scala/meta/internal/implementation/SuperMethodProvider.scala

```diff
@@ -0,0 +1,280 @@
+package scala.meta.internal.implementation
+
+import scala.meta.internal.semanticdb.Scala._
+import scala.meta.internal.semanticdb.ClassSignature
+import scala.meta.internal.semanticdb.MethodSignature
+import scala.meta.internal.semanticdb.SymbolInformation
+import scala.meta.internal.semanticdb.SymbolOccurrence
+import scala.meta.internal.semanticdb.TextDocument
+import scala.meta.internal.semanticdb.TypeRef
+import scala.meta.internal.semanticdb.TypeSignature
+import scala.collection.{mutable => m}
+import scala.meta.internal.metals.codelenses.SuperMethodLensesProvider.LensGoSuperCache
+
+object SuperMethodProvider {
+
+  def findSuperForMethodOrField(
+      methodSymbolInformation: SymbolInformation,
+      documentWithPath: TextDocumentWithPath,
+      symbolRole: SymbolOccurrence.Role,
+      findSymbol: String => Option[SymbolInformation],
+      cache: LensGoSuperCache
+  ): Option[String] = {
+    if (isDefinitionOfMethodField(symbolRole, methodSymbolInformation)) {
+      findSuperForMethodOrFieldChecked(
+        methodSymbolInformation,
+        methodSymbolInformation.signature.asInstanceOf[MethodSignature],
```

Comment:
Can we remove all usage of `asInstanceOf`? The idiomatic way to do this here would be to use pattern matching
```scala
info.signature match {
  case m: MethodSignature =>
    ...
    Some(...)
  case _ =>
    None
```

Reply:
Okay, moved this to `isDefinitionOfMethodField` which returns `Option` and refactored as `for` comprehension. Same with similar situation below :+1: